dali/operators/segmentation/utils/searchable_rle_mask_test.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "dali/operators/segmentation/utils/searchable_rle_mask.h"
+
+namespace dali {
+
+TEST(SearchableRLEMask, tests) {
+  uint8_t mask[] = {
```

Comment:
Can you (in addition, not instead of this one) test a pattern that starts and ends with a 1? Specifically, to check that 0 and n-1 indices are handled correctly.

Reply:
have you seen the extra tests below? (starts at line 62)